internal/services/mssqlmanagedinstance/mssql_managed_instance_start_stop_schedule_resource.go

```diff
@@ -0,0 +1,352 @@
+	Name                 string              `tfschema:"name"`
+	SqlManagedInstanceId string              `tfschema:"managed_instance_id"`
+	Description          string              `tfschema:"description"`
+	ScheduleList         []ScheduleItemModel `tfschema:"schedule_list"`
+	TimeZoneId           string              `tfschema:"timezone_id"`
+	NextExecutionTime    string              `tfschema:"next_execution_time"`
+	NextRunAction        string              `tfschema:"next_run_action"`
+}
+
+type ScheduleItemModel struct {
+	StartDay  schedule.DayOfWeek `tfschema:"start_day"`
+	StartTime string             `tfschema:"start_time"`
+	StopDay   schedule.DayOfWeek `tfschema:"stop_day"`
+	StopTime  string             `tfschema:"stop_time"`
+}
+
+type MsSqlManagedInstanceStartStopScheduleResource struct{}
+
+var _ sdk.ResourceWithUpdate = MsSqlManagedInstanceStartStopScheduleResource{}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) ResourceType() string {
+	return "azurerm_mssql_managed_instance_start_stop_schedule"
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) ModelObject() interface{} {
+	return &SqlManagedInstanceStartStopScheduleModel{}
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.ManagedInstanceStartStopScheduleID
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"managed_instance_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSqlManagedInstanceID,
+		},
+
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"schedule_list": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"start_day": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(schedule.PossibleValuesForDayOfWeek(), false),
+					},
+
+					"start_time": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"stop_day": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(schedule.PossibleValuesForDayOfWeek(), false),
+					},
+
+					"stop_time": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"timezone_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Default:      "UTC",
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
```

Comment:
It's not necessary to output these fields.

Reply:
Updated